File: 2st/Riyeon/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    x = board[0]//2
    y = board[1]//2
    for i in keyinput:
        if i=='left' and abs(answer[0]-1)<=x:
            answer[0] -= 1
        elif i=='right' and abs(answer[0]+1)<=x:
            answer[0] += 1
        elif i=='up' and abs(answer[1]+1)<=y:
            answer[1] += 1
        elif i=='down' and abs(answer[1]-1)<=y:
            answer[1] -= 1
    return answer

File: 4st/Eunseong/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        for (int i = 0; i < keyinput.length; i++) {
            switch (keyinput[i]) {
                case "left" : answer[0] -= 1; break;
                case "right" : answer[0] += 1; break;
                case "up" : answer[1] += 1; break;
                case "down" : answer[1] -= 1; break;
            }
            if (answer[0] < board[0] / -2 || answer[0] > board[0] / 2) {
                answer[0] = (answer[0] < 0) ? answer[0] + 1 : answer[0] - 1;
            } else if (answer[1] < board[1] / -2 || answer[1] > board[1] / 2) {
                answer[1] = (answer[1] < 0) ? answer[1] + 1 : answer[1] - 1;
            }
        }
        return answer;
    }
}